Modules/_elementtree.c

```diff
@@ -850,8 +852,7 @@ _elementtree_Element___deepcopy___impl(ElementObject *self, PyObject *memo)
         if (element_resize(element, self->extra->length) < 0)
             goto error;
 
-        // TODO(picnixz): check for an evil child's __deepcopy__ on 'self'
-        for (i = 0; i < self->extra->length; i++) {
+        for (i = 0; self->extra && i < self->extra->length; i++) {
```

Comment:
Add ` && i < element->extra->length`. The original element can grow in process.

Or call `element_resize()` if `element->extra->length < self->extra->length` before assigning to `element->extra->children[i]`.

We can get duplicated attributes with the same name in the resulting Element, but at least it will not crash or leak.